qutip/solver/result.py

```diff
@@ -982,6 +986,167 @@ def runs_photocurrent(self):
+        weighting the jump and no-jump trajectories
+        """
+        for i, k in enumerate(self._raw_ops):
+            expect_traj = trajectory.expect[i]
+            p = self.no_jump_prob
+            if self.num_trajectories == 1:
+                self._sum_expect_no_jump[i] += expect_traj * p
+                self._sum2_expect_no_jump[i] += expect_traj**2 * p
+                # no jump trajectory will always be the first one, no need
+                # to worry about including jump trajectories
+                avg = self._sum_expect_no_jump[i]
+                avg2 = self._sum2_expect_no_jump[i]
+            else:
+                self._sum_expect_jump[i] += expect_traj * (1 - p)
+                self._sum2_expect_jump[i] += expect_traj**2 * (1 - p)
+                avg = (self._sum_expect_no_jump[i]
+                       + self._sum_expect_jump[i]
+                       / (self.num_trajectories - 1))
+                avg2 = (self._sum2_expect_no_jump[i]
+                        + self._sum2_expect_jump[i]
+                        / (self.num_trajectories - 1))
+
+            self.average_e_data[k] = list(avg)
+
+            # mean(expect**2) - mean(expect)**2 can something be very small
+            # negative (-1e-15) which raise an error for float sqrt.
+            self.std_e_data[k] = list(np.sqrt(np.abs(avg2 - np.abs(avg**2))))
+
+            if self.runs_e_data:
+                self.runs_e_data[k].append(trajectory.e_data[k])
+
+    @property
+    def average_states(self):
+        """
+        States averages as density matrices.
+        """
+        if self._sum_states_no_jump is None:
+            return None
+        p = self.no_jump_prob
+        return [p * final_no_jump
+                + (1 - p) * final_jump / (self.num_trajectories - 1)
+                for final_no_jump, final_jump in
+                zip(self._sum_states_no_jump, self._sum_states_jump)]
+
+    @property
+    def average_final_state(self):
+        """
+        Last states of each trajectory averaged into a density matrix.
+        """
+        if self._sum_final_states_no_jump is None:
+            return None
+        p = self.no_jump_prob
+        return (
+            p * self._sum_final_states_no_jump
+            + (1 - p) * self._sum_final_states_jump
+            / (self.num_trajectories - 1)
+        )
+
+    def __add__(self, other):
+        raise NotImplemented
+
+    @property
+    def photocurrent(self):
+        """
+        Average photocurrent or measurement of the evolution.
+        """
+        cols = [[] for _ in range(self.num_c_ops)]
+        tlist = self.times
+        for collapses in self.collapse:
+            for t, which in collapses:
+                cols[which].append(t)
+        mesurement = [
+            (1 - self.no_jump_prob) *
+            np.histogram(cols[i], tlist)[0] / np.diff(tlist) / (self.num_trajectories - 1)
+            for i in range(self.num_c_ops)
+        ]
```

Comment:
Line too long otherwise:
```suggestion
        mesurement = [
            (1 - self.no_jump_prob) / (self.num_trajectories - 1) *
            np.histogram(cols[i], tlist)[0] / np.diff(tlist)
            for i in range(self.num_c_ops)
        ]
```